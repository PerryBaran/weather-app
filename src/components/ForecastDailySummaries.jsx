import React from "react";
import "../styles/ForecastDailySummaries.css";
import PropTypes from "prop-types";
import ForecastSummary from "./ForecastDailySummary";
import { calcMean } from "../helpers/calculateValues";

function ForecastDailySummaries(props) {
  const { forecasts, handleForecastSelect } = props;

  const getDailyForecast = (forecastsArray) => {
    return forecastsArray.map((forecast) => {
      const [{ date, dateTime, icon, weather }] = forecast;
      return {
        date: date.slice(0, 10),
        dateTime,
        icon,
        temp: calcMean(forecast, "temp", 2),
        weather,
      };
    });
  };

  const dailyForecast = getDailyForecast(forecasts);

  return (
    <div className="forecast-daily-summaries">
      {dailyForecast.map((forecast) => {
        const { dateTime, date, weather, icon, temp } = forecast;
        return (
          <ForecastSummary
            dateTime={dateTime}
            date={date}
            weather={weather}
            icon={icon}
            temp={temp}
            handleForecastSelect={handleForecastSelect}
            key={forecast.dateTime}
          />
        );
      })}
    </div>
  );
}

ForecastDailySummaries.propTypes = {
  forecasts: PropTypes.arrayOf(
    PropTypes.arrayOf(
      PropTypes.shape({
        date: PropTypes.string.isRequired,
        dateTime: PropTypes.number.isRequired,
        icon: PropTypes.string.isRequired,
        temp: PropTypes.number.isRequired,
        weather: PropTypes.string.isRequired,
      })
    )
  ).isRequired,
  handleForecastSelect: PropTypes.func.isRequired,
};

export default ForecastDailySummaries;
