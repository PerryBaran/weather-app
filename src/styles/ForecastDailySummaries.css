.forecast-daily-summaries {
  display: flex;
  justify-content: space-evenly;
}