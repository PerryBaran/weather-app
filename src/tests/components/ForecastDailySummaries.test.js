import React from "react";
import { render } from "@testing-library/react";
import ForecastDailySummaries from "../../components/ForecastDailySummaries";

describe("ForecastDailySummaries", () => {
  const validProps = {
    forecasts: [
      [
        {
          date: "today",
          dateTime: 123,
          icon: "200",
          temp: 12,
          weather: "really cold",
        },
        {
          date: "today",
          dateTime: 456,
          icon: "900",
          temp: 14,
          weather: "not cold",
        },
      ],
      [
        {
          date: "tomorrow",
          dateTime: 789,
          icon: "800",
          temp: 13,
          weather: "kinda cold",
        },
        {
          date: "tomorrow",
          dateTime: 101,
          icon: "600",
          temp: 15,
          weather: "still cold",
        },
      ],
    ],
    handleForecastSelect: () => {},
  };

  describe("snapshot", () => {
    it("renders correctlty", () => {
      const { asFragment } = render(
        <ForecastDailySummaries
          forecasts={validProps.forecasts}
          handleForecastSelect={validProps.handleForecastSelect}
        />
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe("list", () => {
    it("renders correct values for props", () => {
      const { getAllByTestId } = render(
        <ForecastDailySummaries
          forecasts={validProps.forecasts}
          handleForecastSelect={validProps.handleForecastSelect}
        />
      );

      expect(getAllByTestId("forecast-daily-summary")).toHaveLength(
        validProps.forecasts[0].length
      );
    });
  });
});
