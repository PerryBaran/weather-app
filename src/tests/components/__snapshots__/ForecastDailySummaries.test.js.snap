// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDailySummaries snapshot 1`] = `
<DocumentFragment>
  <div
    class="forecast-daily-summaries"
  >
    <div
      class="forecast-daily-summary"
      data-testid="forecast-daily-summary"
    >
      <p>
        Thu
        <span
          class="forecast-daily-summary__line-break"
        >
           Nov 03
        </span>
      </p>
      <div
        class="forecast-daily-summary__icon"
        data-testid="forecast-icon"
      >
        <i
          class="wi wi-day-storm-showers"
          name="owm"
        />
      </div>
      <p>
        unique1
        <span
          class="forecast-daily-summary__line-break"
        >
          weather1
        </span>
      </p>
      <p>
        10.5°C
      </p>
      <button
        class="forecast-daily-summary__more-details"
        type="button"
      >
        More Details
      </button>
    </div>
    <div
      class="forecast-daily-summary"
      data-testid="forecast-daily-summary"
    >
      <p>
        Tue
        <span
          class="forecast-daily-summary__line-break"
        >
           Nov 08
        </span>
      </p>
      <div
        class="forecast-daily-summary__icon"
        data-testid="forecast-icon"
      >
        <i
          class="wi wi-day-sunny"
          name="owm"
        />
      </div>
      <p>
        unique3
        <span
          class="forecast-daily-summary__line-break"
        >
          weather3
        </span>
      </p>
      <p>
        12.5°C
      </p>
      <button
        class="forecast-daily-summary__more-details"
        type="button"
      >
        More Details
      </button>
    </div>
    <div
      class="forecast-daily-summary"
      data-testid="forecast-daily-summary"
    >
      <p>
        Sun
        <span
          class="forecast-daily-summary__line-break"
        >
           Feb 26
        </span>
      </p>
      <div
        class="forecast-daily-summary__icon"
        data-testid="forecast-icon"
      >
        <i
          class="wi wi-day-snow"
          name="owm"
        />
      </div>
      <p>
        unique5
        <span
          class="forecast-daily-summary__line-break"
        >
          weather5
        </span>
      </p>
      <p>
        14.5°C
      </p>
      <button
        class="forecast-daily-summary__more-details"
        type="button"
      >
        More Details
      </button>
    </div>
  </div>
</DocumentFragment>
`;
